### DISCUSSION TITLE
Support web components

### DISCUSSION THREAD

**Discussion by khanh-le-otsv** [chakra-ui/ark#697] [Ideas]
First just want to say thank you for making such great library. Second I want to suggest an idea about supporting web components. React 19 will support web components, vue already supports them. So adding web components would bring a huge benefit

**Comment by cschroeter** [comment]
Thank you for your kind words and suggestion!

We're glad to hear that you're excited about our library.  We plan to support a wide range of modern UI frameworks like React, Solid, Angular, Svelte, and more. We believe supporting all of this frameworks is essential to make our library accessible and useful to as many developers as possible.

Regarding your suggestion about adding support for web components, we completely agree that it would be a great addition to our library. We're actually working on a roadmap for our library that we plan to share later in April, and we'll definitely keep your suggestion in mind as we continue to develop and improve our library.

Thanks again for your feedback and support!

**Reply by vhscom** [reply]
Headless UI eventually got a Svelte community port. And imitation is the sincerest form of flattery. What I see in Ark looks very nice and I personally wouldn't mind seeing this radio activated for selfish reasons. 😉![Screenshot_20231228_002212.jpg](https://github.com/chakra-ui/ark/assets/97140109/8a18d755-7f19-440e-8a3f-d3630ff96113)

**Comment by cschroeter** [comment]
With Svelte 5+ there is a realistic chance that this can happen. For more information about Svelte: https://github.com/chakra-ui/zag/discussions/262

**Reply by Omikorin** [reply]
I am sane enough to implement components in Svelte too :D
